resources/custom-resources/customize-cognito-ui/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/CustomizeCognitoUi.java

```diff
@@ -0,0 +1,121 @@
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.cognitoidentityprovider.CognitoIdentityProviderClient;
+import software.amazon.awssdk.services.cognitoidentityprovider.model.CognitoIdentityProviderException;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+public class CustomizeCognitoUi implements RequestHandler<Map<String, Object>, Object> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CustomizeCognitoUi.class);
+    private static final String AWS_REGION = System.getenv("AWS_REGION");
+    private final CognitoIdentityProviderClient cognito;
+    private final S3Client s3;
+
+    public CustomizeCognitoUi() {
+        if (Utils.isBlank(AWS_REGION)) {
+            throw new IllegalStateException("Missing required environment variable AWS_REGION");
+        }
+        LOGGER.info("Version Info: {}", Utils.version(this.getClass()));
+        this.cognito = Utils.sdkClient(CognitoIdentityProviderClient.builder(),
+            CognitoIdentityProviderClient.SERVICE_NAME);
+        this.s3 = Utils.sdkClient(S3Client.builder(), S3Client.SERVICE_NAME);
+    }
+
+    @Override
+    public Object handleRequest(Map<String, Object> event, Context context) {
+        Utils.logRequestEvent(event);
+
+        final String requestType = (String) event.get("RequestType");
+        final Map<String, Object> resourceProperties = (Map<String, Object>) event.get("ResourceProperties");
+        final String logoS3Bucket = (String) resourceProperties.get("LogoS3Bucket");
+        final String logoS3Key = (String) resourceProperties.get("LogoS3Key");
+        final String userPoolId = (String) resourceProperties.get("UserPoolId");
+
+        ExecutorService service = Executors.newSingleThreadExecutor();
+        Map<String, Object> responseData = new HashMap<>();
+        try {
+            Runnable r = () -> {
+                if ("Create".equalsIgnoreCase(requestType) || "Update".equalsIgnoreCase(requestType)) {
+                    LOGGER.info("CREATE or UPDATE");
+                    try {
+                        // get bytes of logo from S3
+                        ResponseInputStream<GetObjectResponse> responseInputStream = s3.getObject(request -> request
+                                .bucket(logoS3Bucket)
+                                .key(logoS3Key)
+                                .build());
+                        SdkBytes logoBytes = SdkBytes.fromInputStream(responseInputStream);
+                        // set ui customization
+                        cognito.setUICustomization(request -> request
+                                .userPoolId(userPoolId)
+                                .imageFile(logoBytes)
+                                .css(".banner-customizable {background-color: blue;}")
```

Comment:
Not fully up-to-speed on the CSS Cognito provides for customization, but wouldn't we want this to match the primary background color in the React app? That looks to be `#321fdb` or `rgb(50, 31, 219)`.